src/autopas/utils/ArrayUtils.h

```diff
@@ -146,4 +147,79 @@ std::enable_if_t<autopas::utils::ArrayUtils::is_container<Container>::value, std
 
   return os;
 }
+
+/**
+ * Given a collection of vectors, redistributes the elements of the vectors so they all have the same (or +1) size.
+ * @tparam OuterContainerT Collection type
+ * @param vecvec Reference to the collection of vectors to be balanced in place.
+ */
+template <class OuterContainerT>
+void balanceVectors(OuterContainerT &vecvec) {
+  balanceVectors(
+      vecvec, [](auto &innerContainer) -> auto & { return innerContainer; });
+}
+
+/**
+ * Given a collection of containers, redistributes the elements of the vectors so they all have the same (or +1) size.
+ * @tparam OuterContainerT Collection type
+ * @tparam F Type of the function innerContainerToVec
+ * @param vecvec Reference to the collection of vectors to be balanced in place.
+ * @param innerContainerToVec Function to map inner containers to std::vector&.
+ */
+template <class OuterContainerT, class F>
+void balanceVectors(OuterContainerT &vecvec, F innerContainerToVec) {
+  using InnerContainerT = typename OuterContainerT::value_type;
+  using ElemT = typename std::remove_reference_t<
+      std::invoke_result_t<decltype(innerContainerToVec), InnerContainerT &>>::value_type;
+  // calculate vecvec statistics
+  const auto vecvecSize = vecvec.size();
+  if (vecvecSize == 0) {
+    // nothing to do
+    return;
+  }
+  const size_t numElem = std::transform_reduce(vecvec.begin(), vecvec.end(), 0, std::plus<>(),
+                                               [&](auto &vec) { return innerContainerToVec(vec).size(); });
+  const auto targetSize = static_cast<long>(numElem / vecvecSize);
+  auto rest = numElem % vecvecSize;
```

Comment:
`rest` is not a clear enough name

Reply:
done 6685dbbb05334e01eb1001ad4b603bcceed1b05b